pennylane/operation.py

```diff
@@ -1132,6 +1131,28 @@ def arithmetic_depth(self) -> int:
         """Arithmetic depth of the operator."""
         return 0
 
+    def map_wires(self, wire_map: dict):
+        """Returns a copy of the current operator with its wires changed according to the given
+        wire map.
+
+        Args:
+            wire_map (dict): dictionary containing the old wires as keys and the new wires as values
+
+        Returns:
+            .Operator: new operator
+        """
+        if any(wire not in wire_map for wire in self.wires):
+            raise ValueError("All operator wires are not present in the given wire map.")
+
+        new_wires = wire_map.values()
```

Comment:
I would interpret "new_wires" as the wires that the new operation will have, 
`Wires([wire_map[wire] for wire in self.wires])`. We also only really care about identical targets in that subset.

```
qml.S(0).map_wires({0:"a", 1:"b", 2:"c"})
```

Reply:
Do you prefer `mapped_wires` instead?

Reply:
> We also only really care about identical targets in that subset.

Then you want to allow the following call?
```python
qml.S(0).map_wires({0:"a", 1:"b", 2:"b"})
```

Reply:
I don't see anything wrong with the above call. We just want to avoid situations like `CNOT((0,0))`.

Reply:
Okay, then I will only check for duplicated target wires in the Operator's wires subset

Reply:
Done!